server/gokbg3/grails-app/domain/org/gokb/cred/TitleInstancePackagePlatform.groovy

```diff
@@ -541,30 +800,28 @@ class TitleInstancePackagePlatform extends KBComponent {
       def title_info = tipp_dto.title
 
       if (title_info instanceof Map) {
-        ti = TitleInstance.get(title_info.id ?: title_info.internalId)
-      } else {
-        ti = TitleInstance.get(title_info)
+        ti = title_info
       }
-
-      log.debug("Title lookup: ${ti}")
     }
 
     def status_current = RefdataCategory.lookupOrCreate(RCConstants.KBCOMPONENT_STATUS, 'Current')
     def status_retired = RefdataCategory.lookupOrCreate(RCConstants.KBCOMPONENT_STATUS, 'Retired')
     def trimmed_url = tipp_dto.url ? tipp_dto.url.trim() : null
+    //TODO: Moe
     def curator = pkg?.curatoryGroups?.size() > 0 ? (user.adminStatus || user.curatoryGroups?.id.intersect(pkg?.curatoryGroups?.id)) : true
 
     if (pkg && plt && ti && curator) {
       log.debug("See if we already have a tipp")
-      def tipps = TitleInstancePackagePlatform.executeQuery('select tipp from TitleInstancePackagePlatform as tipp, Combo as pkg_combo, Combo as title_combo, Combo as platform_combo  ' +
+      def tipps = TitleInstancePackagePlatform.executeQuery('select tipp from TitleInstancePackagePlatform as tipp, Combo as pkg_combo, Combo as platform_combo  ' +
         'where pkg_combo.toComponent=tipp and pkg_combo.fromComponent=?' +
         'and platform_combo.toComponent=tipp and platform_combo.fromComponent = ?' +
-        'and title_combo.toComponent=tipp and title_combo.fromComponent = ?',
-        [pkg, plt, ti])
+        'and (tipp.name = ? or tipp.name = ?)',
```

Comment:
tipp.name in (:tiName,:tiDtoName); avoid using ? in queries (unless ? is performed in separate refactoring step)